### PR TITLE
Support dilated convolutions in ChainerX

### PR DESCRIPTION
**Pull request by take-cheeze**

### CODE REVIEW

File: chainerx_cc/chainerx/cuda/cuda_conv_test.cc [hunk 1]
```diff
@@ -62,9 +62,9 @@ TEST(CudaConvTest, FwdAlgoCache) {
         bool cover_all = false;
 
         EXPECT_EQ(size_t{0}, cuda_internal::CudaConvTest::GetFwdAlgoCacheMapSize(cuda_conv));
-        cuda_conv.Conv(device, x, w, b, stride, pad, cover_all, x.dtype());
+        cuda_conv.Conv(device, x, w, b, stride, pad, {1, 1}, cover_all, x.dtype());
```
Comment (hvy) [inline chainerx_cc/chainerx/cuda/cuda_conv_test.cc]: A nit but how about not inlining "dilate" to be consistent with `stride`, `pad`, etc. and ditto for remaining tests. It's a detail so please ignore it if you disagree.

File: chainerx_cc/chainerx/native/native_device/pool.cc [hunk 2]
```diff
@@ -57,6 +57,14 @@ Axes GetSwapSpatialDimensionsAxes(size_t n) {
     return axes;
 }
 
+StackVector<int64_t, kMaxNdim> GetDilate(size_t ndim) {
+    StackVector<int64_t, kMaxNdim> ret;
```
Comment (hvy) [inline chainerx_cc/chainerx/native/native_device/pool.cc]: How about asserting that `ndim <= kMaxNdim`?

File: tests/chainerx_tests/unit_tests/routines_tests/test_connection.py [hunk 3]
```diff
@@ -102,7 +102,9 @@ def forward_chainerx(self, inputs):
             (x, w), b = inputs, None
         else:
             x, w, b = inputs
-        y = chainerx.conv(x, w, b, self.stride, self.pad, self.cover_all)
+        y = chainerx.conv(
+            x, w, b, stride=self.stride, pad=self.pad, dilate=1,
+            cover_all=self.cover_all)
```
Comment (hvy) [inline tests/chainerx_tests/unit_tests/routines_tests/test_connection.py]: Could you parameterize dilate in these tests?

**Comment by hvy** [comment]
I don't mean to stress and just want to check but are you now working on the test parameterization?

**Comment by take-cheeze** [comment]
@hvy I'm trying on it but extending the parameter of convolution is quite difficult.
Thanks for reminding me.

**Comment by take-cheeze** [comment]
Seems things are working except style checking. Will fix it

**Comment by hvy** [comment]
Jenkins, test this please.

**Comment by hvy** [comment]
I have not investigated the cause yet but I tried running the tests 1000 times and the `ConvTranspose` backward failed 7 times. Out dtype was `float16`, `cuda` and `dilate=1`. The error seems to go beyond numerical tolerances.

**Comment by take-cheeze** [comment]
@hvy Thank you for investigation! I'll check for it!

**Pull request closed without merging**